mautic/mautic
# Fix for error caused by null date_left in template

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Yes]
| New feature/enhancement? (use the a.x branch)      | [No]
| Deprecations?                          | [No]
| BC breaks? (use the c.x branch)        | [No]
| Automated tests included?              | [No]
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Description:

I get a 500 error in the contact view when the contact has at least one hit page with null as date left value. PR introduces additional condition that checks whether the left date value is not empty before proceeding with the calculation of timeOnPage.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open a preview (s/contacts/view/{id}) for a contact who has visited one of the pages or forms, but has null in date left in page hit.
3. Without "is not empty" condition, the following error should appear in the logs::
```mautic.CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "Impossible to access an attribute ("timestamp") on a null variable." at /var/www/html/app/bundles/PageBundle/Views/SubscribedEvents/Timeline/index.html.twig line 5 {"exception":"[object] (Twig\\Error\\RuntimeError(code: 0): Impossible to access an attribute (\"timestamp\") on a null variable. at /var/www/html/app/bundles/PageBundle/Views/SubscribedEvents/Timeline/index.html.twig:5)"} {"hostname":"mautic-web","pid":2007}```
